### DISCUSSION TITLE
disable scan request responses

### DISCUSSION THREAD

**Discussion by ReggiePark** [pvvx/ATC_MiThermometer#483] [Ideas]
Hi,
I don't know the Bluetooth protocol that well but is it possible to tell other devices not to send scan request? 
It would be nice to have this as an option (or just add an option to not respond to scan requests) if all you need is the advertised data since it could extend the battery life a lot, especially if you are in an area with lots of other devices constantly bombarding you with requests.

Thanks,
RP

**Comment by pvvx** [comment]
Possible, but the device name will not be scanned.
And problems may arise in Linux. Everything is bad with BLE there.
It's easier to use passive scanning.
Current, already implemented, functions in firmwares ATC_MiThermometer  are 9 years ahead of Linux capabilities in BLE.
For these reasons, further additions of options and functions have been stopped. We are waiting for the situation with BLE in Linux to be fixed.
A similar situation occurs in the "[Web Bluetooth API](https://github.com/WebBluetoothCG/web-bluetooth/blob/main/implementation-status.md#scanning-api)". Everything is stopped and waiting for Linux patchs.

**Reply by ReggiePark** [reply]
I don't need the name, I use the MAC address, since they all start the same way.

Of course, I see your point regarding Linux problems and I guess, even if I don't need it, it may cause problems on some other devices.

I already use passive scanning but it's all those other devices draining the battery with scan requests...

RP

**Reply by pvvx** [reply]
Bluez on Linux only works with Bluetooth version 4.2. Since 2015 this is the BLE "Legacy" type. That is, only an outdated format.
As a result, only `Undirected advertising` is accepted (in HA).

Find here this type of BLE advertising so that it is accepted in Bluez:

https://novelbits.io/bluetooth-low-energy-advertisements-part-1/

## Legacy Advertising PDUs

These are available for all Bluetooth versions – also enables backward compatibility with older versions and are used on the Primary advertising channels.

ADV_IND: Connectable Scannable Undirected advertising.
ADV_DIRECT_IND: Connectable Directed advertising
ADV_NONCONN_IND: Non-Connectable Non-Scannable Undirected advertising
ADV_SCAN_IND: Scannable Undirected advertising

## Extended Advertising PDUs

These types were introduced in Bluetooth version 5.0. They provide devices the option to advertise on the Secondary advertising channels in addition to the Primary advertising channels. The benefit of the Secondary channels is that they allow increased advertising data capacity.

ADV_EXT_IND: Extended advertising (used for all advertising types except Connectable Scannable Undirected) – sent on the Primary advertising channels.
AUX_ADV_IND: Extended advertising (used for all advertising types except Connectable Scannable Undirected) – sent on the Secondary advertising channels.
AUX_SYNC_IND: used for Periodic advertising
AUX_CHAIN_IND: used with other advertising types to hold additional advertising data (chaining of advertising packets)

**Reply by ReggiePark** [reply]
Doesn't that almost look like it would be possible without causing problems?
'Undirected advertising' is still want I want, just with the ADV_NONCONN_IND flag. The problem is that it would leave it in a non-connectable state, but this could be solved by, say, using the ADV_IND flag for the first hour after the power was connected, once the option is active.

RP

**Reply by pvvx** [reply]
Source codes are provided - build your own version and experiment with Home Assistant.

`nRFConnect` for smartphone can broadcast designated BLE advertisements.

**Reply by ReggiePark** [reply]
Already working on my own version ;-) 

Why Home Assistant?

RP

**Reply by pvvx** [reply]
> Why Home Assistant?

These thermometers are most often used in HA.

**Reply by ReggiePark** [reply]
I thought it something more technical, I use my own software...

RP

**Reply by pvvx** [reply]
> I thought it something more technical, I use my own software...

Then what are the problems? From normal CR2032 Xiaomi LYWSD03MMC will work for a year at home temperatures.

**Reply by ReggiePark** [reply]
No problems, just want it to last longer, just for fun, maybe two years...

RP

**Reply by pvvx** [reply]
The difference in consumption between different hardware versions will be greater than with frequent active scan polling...
Two years with the display turned on will not work. The controller sleep current is 3..4 times less than the display consumes.

Total from 9 µA - And this amount cannot provide 2 years of operation from a cheap CR2032.
The actual capacity of CR2032 batteries varies by more than 5 times. Depends on the manufacturer and price.

![image](https://github.com/pvvx/ATC_MiThermometer/assets/12629515/6f9fa59a-f1b8-4eb1-9f3b-1931ce697dfe)

Considering the price of CR2032 batteries at $0.1, there is no point in fishing for -10%, because Soldering the capacitor into an empty space on the printed circuit board gives +40% additional operating time.

And taking into account everything described above, no one will notice the difference from disabling the scan response.

Implementation of disabling the reaction to connection will be possible later, but for thermometers with a button. To protect against unauthorized connection.
The scaning will still remain.

**Reply by ReggiePark** [reply]
I didn't literally mean two years (of course, you could always hope), it was just the thought that turning off this additional activity should result in a better battery life.

Should it be possible to turn off the display of the LYWSD03MMC device? I checked the box on the configuration page but nothing happened, even after a power cycle.

Also, what capacitor are you talking about?

RP

**Reply by ReggiePark** [reply]
PS.: if you want to protect against unauthorized connections, you might still consider the option to only allow a connection during the first 15 (or so) minutes after a power cycle for devices without a button...

**Reply by pvvx** [reply]
> Also, what capacitor are you talking about?

Capacitors that are provided on the printed circuit board by developers. But they were not soldered in production. Xiaomi saved a couple of cents here by forcing you to change batteries more often.

**Reply by ReggiePark** [reply]
PPS.: regarding the LCD-screen, I noticed from the code that it is just not updating it anymore. I was expecting either a blank screen or better yet, a smiley instead of a frozen screen, to indicate that the device is on but just not updating the screen.

**Reply by pvvx** [reply]
Post a technical description of the LCD controller for your thermometer model.
On other well-known models, the screen turns off completely.

**Reply by ReggiePark** [reply]
I'm not fit to live...
After your message I removed the battery and put it back in, and, like you said, the screen was blank. I was absolutely sure that I did this before, since it was the obvious thing to do.

My apologies.

RP
